File: code/warum.cs
using System;
using System.Collections.Generic;

public class HelloWorld
{
  public static List<int> cache;
  public static void doSomething(List<int> list) { //null?
    list.Add(5);  // no language-level const
    cache = list; // shareability is a type-design decision
  }
  
  public static void main(String[] args)
  {
    List<int> my_list = new List<int>();
    my_list.Add(1);
    doSomething(my_list);
    doSomething(null);
  }
}

File: code/warum.java
import java.util.ArrayList;

public class HelloWorld
{
  static ArrayList<Integer> cache;
  static void doSomething(ArrayList<Integer> list) { // null?
    list.add(5);  // no language-level const
    cache = list; // (almost) everything is always shared 
  }
  
  public static void main(String[] args)
  {
    final ArrayList<Integer> my_list = new ArrayList<>();
    my_list.add(1);
    doSomething(my_list);
    doSomething(null);
  }
}

